actionpack/test/dispatch/request_id_test.rb

```diff
@@ -75,7 +84,9 @@ def with_test_route_set(header: "X-Request-Id")
         end
 
         @app = self.class.build_app(set) do |middleware|
+          middleware.use Rack::Lint
           middleware.use ActionDispatch::RequestId, header: header
+          middleware.use Rack::Lint
```

Comment:
Just curious to know that, Why do we need to use Rack::Lint middleware before and after the ActionDispatch::RequestId?

Reply:
It's advised by Rack::Lint: https://github.com/rack/rack/blob/444dc8a130663159ed490f25948b73ff9b5fbba0/lib/rack/lint.rb#L27-L28

I believe this helps to ensure that the rack app, request and response are valid before it reaches your middleware, and is still valid once your middleware returns.

Reply:
Exactly, the goal is to catch errors both going down the stack as well as up. The `Rack::Lint` after ensures that `RequestId` doesn't make any changes to the Request `env` that don't follow SPEC, and the `Rack::Lint` before ensures that `RequestId` doesn't make any changes to the Response that don't follow the SPEC.